api/client/commands.go

```diff
@@ -2660,6 +2666,10 @@ func (s *containerStats) Collect(cli *DockerCli) {
 		for {
 			var v *types.Stats
 			if err := dec.Decode(&v); err != nil {
+				if err == io.EOF && !streamStats {
```

Comment:
Ok sorry, I misunderstood on first read. So this is basically saying: `EOF` is expected when you didn't ask for `streamStats`.